src/lib/network_pool/snark_pool.ml

```diff
@@ -386,23 +406,35 @@ module Make (Transition_frontier : Transition_frontier_intf) :
         Transaction_snark_work.Checked.create_unsafe
           {Transaction_snark_work.fee; proofs= proof; prover} )
 
+  let store_periodically (t : Resource_pool.t) =
+    Clock.every' (Time.Span.of_min 3.) (fun () ->
+        let before = Time.now () in
+        let%map () =
+          Writer.save_bin_prot t.config.disk_location
+            Snark_tables.Serializable.Stable.Latest.bin_writer_t
+            (Snark_tables.to_serializable t.snark_tables)
+        in
+        let elapsed = Time.diff (Time.now ()) before in
+        [%log' debug t.logger] "SNARK pool serialization took $time ms"
+          ~metadata:[("time", `Float (Time.Span.to_ms elapsed))] )
+
   let load ~config ~logger ~constraint_constants ~consensus_constants
-      ~time_controller ~disk_location ~incoming_diffs ~local_diffs
-      ~frontier_broadcast_pipe =
+      ~time_controller ~incoming_diffs ~local_diffs ~frontier_broadcast_pipe =
     let tf_diff_reader, tf_diff_writer =
       Strict_pipe.(
         create ~name:"Snark pool Transition frontier diffs" Synchronous)
     in
     match%map
-      Async.Reader.load_bin_prot disk_location
-        Snark_tables.Stable.Latest.bin_reader_t
+      Async.Reader.load_bin_prot config.Resource_pool.Config.disk_location
+        Snark_tables.Serializable.Stable.Latest.bin_reader_t
     with
     | Ok snark_table ->
         let pool = Resource_pool.of_serializable snark_table ~config ~logger in
         let network_pool =
           of_resource_pool_and_diffs pool ~logger ~constraint_constants
             ~incoming_diffs ~local_diffs ~tf_diffs:tf_diff_reader
         in
+        store_periodically (resource_pool network_pool) ;
```

Comment:
Seems fine given that we only ever create the snark pool once during the lifetime of the process, but you may want to add a comment to the load function here stating that it this effectively makes the snark pool never GC and that you cannot make more than one without corrupting the storage. Perhaps we could even add a global bool to enforce this so that we don't end up debugging this one day.

Reply:
:+1: